### ISSUE TITLE
Bug in BIFReader function

### ISSUE THREAD

**Issue by shivani1401** [pgmpy/pgmpy#1053]
### Subject of the issue
There is a bug in the BIFReader function. I think while reading the CPD only the order  in which the states are specified is taken into account , not the actual values of the state.

e.g. variable Pollution {low high} **high - 1 low - 0**
variable Smoker {True False} **True - 0, False-1**

probability ( Cancer | Pollution, Smoker ) {
  (low, True) 0.03, 0.97; 
  (high, True) 0.05, 0.95; **should be 1,0 instead of 0,1**
  (low, False) 0.001, 0.999;**should be 0,1 instead of 1,0**
  (high, False) 0.02, 0.98;
}

probability ( Cancer | Pollution, Smoker ) {
  (low, True) 0.03, 0.97; 
 (low, False) 0.001, 0.999;
  (high, True) 0.05, 0.95;
  (high, False) 0.02, 0.98;
}
The two CPDs are equivalent but read differently by the function.
### Your environment
* pgmpy version 0.1.7
* Python version 3.5
* Operating System Unbuntu16.04

### Steps to reproduce
Attached are two benchmark files cancer1.bif and cancer2.bif with the same data but the order of specifying the states of CPD is changed for one of the tables. Inference on both models gives different results even though the model is the same. 
Sample run file is also attached.

### Expected behaviour
The result of the query should be same in both cases since models are equivalent.

### Actual behaviour
Results with model file1:
```
+----------+---------------+
| Cancer   |   phi(Cancer) |
+==========+===============+
| Cancer_0 |        0.0116 |
+----------+---------------+
| Cancer_1 |        0.9884 |
+----------+---------------+
Results with model file2:
+----------+---------------+
| Cancer   |   phi(Cancer) |
+==========+===============+
| Cancer_0 |        0.0410 |
+----------+---------------+
| Cancer_1 |        0.9590 |
+----------+---------------+
```
[test.zip](https://github.com/pgmpy/pgmpy/files/2775737/test.zip)

**Comment by ankurankan** [comment]
@shivani1401 The int values to the states is given based on when it's encountered. So, if `low` comes before `high`, `low` get the value of 0 and `high` gets 1. With that, the reader seems to be working fine. 

But ideally, we would like to keep the state names instead of assigning integers to them. I will create a new issue for that. Thanks for reporting this.

**Comment by shivani1401** [comment]
So the assignment to variable states is done based on when it is encountered. So, "variable Pollution {low high}" should assign **low - 0, high -1** and "variable Smoker {True False}" should assign **True - 0, False-1**. Now when, the following is encountered probability ( Cancer | Pollution, Smoker ) {
(low, True) 0.03, 0.97;
(high, True) 0.05, 0.95; 
(low, False) 0.001, 0.999;
(high, False) 0.02, 0.98;
}
shouldn't **(high, True)**  be assigned **1,0** instead of 0,1
and **(low, False)** be assigned **0,1** instead of 1,0

**Comment by shivani1401** [comment]
I think the order in which the following four lines are written shouldn't impact the reading once the int assignment to the variable states is done. Please let me know if there is something I am missing on.
(low, True) 0.03, 0.97;
(high, True) 0.05, 0.95;
(low, False) 0.001, 0.999;
(high, False) 0.02, 0.98

Thanks

**Comment by ankurankan** [comment]
@shivani1401 Yeah, this seems to be a bug. I think the current code assumes the order of the values in the way the states are defined and doesn't really look at the states in each individual line. Thanks for pointing this out.

**Comment by ankurankan** [comment]
@shivani1401 This should be fixed now in the latest dev.
